web3.js/src/transaction/legacy.ts

```diff
@@ -726,10 +726,15 @@ export class Transaction {
   }
 
   /**
-   * Verify signatures of a complete, signed Transaction
-   */
-  verifySignatures(): boolean {
-    return this._verifySignatures(this.serializeMessage(), true);
+   * Verify signatures of a complete Transaction
+   * Optional parameter specifies if we're expecting a fully signed Transaction or a partially signed one.
+   * If no boolean is provided, we expect a fully signed Transaction by default.
+   */
+  verifySignatures(requireAllSignatures?: boolean): boolean {
+    return this._verifySignatures(
+      this.serializeMessage(),
+      requireAllSignatures === undefined ? true : requireAllSignatures,
```

Comment:
```suggestion
      !!requireAllSignatures,
```

Reply:
I believe this suggestion is incorrect, as we want to be backwards compatible, meaning that if the caller does not specify any parameter (as is the case in the old version) the function should behave like its old version i.e. only verify fully signed txs. This means if `requireAllSignatures` is `undefined`, it should be true by default, but `!!undefined = false`.

Reply:
Oh right, sorry. My bad!